### PR TITLE
Feature/locations

### PR DESCRIPTION
**Pull request by rma-bryson**
Adding Locations DTO with Controller. Added Location end point class. Added tests.

### CODE REVIEW

**Review by rma-psmorris — CHANGES_REQUESTED** [review]
Rebase onto main please

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LocationController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import mil.army.usace.hec.cwms.http.client.HttpRequestBuilder;
+import mil.army.usace.hec.cwms.http.client.HttpRequestResponse;
+import mil.army.usace.hec.cwms.radar.client.model.Location;
+import mil.army.usace.hec.cwms.radar.client.model.RadarObjectMapper;
+
+
+public class LocationController {
+
+    public Location retrieveLocation(ApiConnectionInfo apiConnectionInfo, LocationEndPointInput locationEndpointInput) throws IOException {
+        HttpRequestResponse response = new HttpRequestBuilder(apiConnectionInfo, "locations")
```
Comment (rma-psmorris) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LocationController.java:38]: "locations" should be a constant

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to deal
  * in the Software without restriction, including without limitation the rights
- * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * to use, copy, modifdy, merge, publish, distribute, sublicense, and/or sell
```
Comment (rma-psmorris) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java:9]: introducing a typo

File: cwms-radar-client/src/test/resources/radar/json/catalog_ts.json [hunk 3]
```diff
@@ -0,0 +1,3007 @@
+{
```
Comment (rma-psmorris) [inline cwms-radar-client/src/test/resources/radar/json/catalog_ts.json:1]: this file isnt related to your change

File: cwms-radar-client/src/test/resources/radar/json/catalog_tspage1.json [hunk 4]
```diff
@@ -0,0 +1,3007 @@
+{
```
Comment (rma-psmorris) [inline cwms-radar-client/src/test/resources/radar/json/catalog_tspage1.json:1]: this file isnt related to your change

File: cwms-radar-client/src/test/resources/radar/json/catalog_tspage2.json [hunk 5]
```diff
@@ -0,0 +1,3007 @@
+{
```
Comment (rma-psmorris) [inline cwms-radar-client/src/test/resources/radar/json/catalog_tspage2.json:1]: this file isnt related to your change

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationEndpointInput.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+package mil.army.usace.hec.cwms.radar.client.controllers;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+
+class TestLocationEndpointInput {
+    @Test
+    void testNullLocationName() {
```
Comment (rma-psmorris) [inline cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationEndpointInput.java:34]: this only tests failure, where is the test of normal operation?

**Review by rma-psmorris — CHANGES_REQUESTED** [review]
review of location endpoint client api

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationEndpointInput.java (hunk 6 above)
Comment (rma-bryson) [reply cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestLocationEndpointInput.java:53]: Added test of normal operations for Location Endpoint input and TimeSeries Endpoint input classes

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LocationController.java (hunk 1 above)
Comment (rma-bryson) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/LocationController.java]: Refactored "locations" into a constant

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java (hunk 2 above)
Comment (rma-bryson) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java]: Fixed typo

File: cwms-radar-client/src/test/resources/radar/json/catalog_tspage2.json (hunk 5 above)
Comment (rma-bryson) [reply cwms-radar-client/src/test/resources/radar/json/catalog_tspage2.json]: Removed this

File: cwms-radar-client/src/test/resources/radar/json/catalog_tspage1.json (hunk 4 above)
Comment (rma-bryson) [reply cwms-radar-client/src/test/resources/radar/json/catalog_tspage1.json]: Removed this.

File: cwms-radar-client/src/test/resources/radar/json/catalog_ts.json (hunk 3 above)
Comment (rma-bryson) [reply cwms-radar-client/src/test/resources/radar/json/catalog_ts.json]: Removed this.

**Review by rma-psmorris — CHANGES_REQUESTED** [review]
one change I think

File: cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OkHttpClientInstance.java [hunk 7]
```diff
@@ -35,7 +35,7 @@
  */
 final class OkHttpClientInstance {
 
-    private static final Logger LOGGER = Logger.getLogger(HttpRequestBuilder.class.getName());
+    private static final Logger LOGGER = Logger.getLogger(HttpRequestBuilderImpl.class.getName());
```
[the remark this answers is not in the document]
Comment (rma-bryson) [reply cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/OkHttpClientInstance.java:38]: yeah, looks like that was wrong from the start. I'll fix it.

**Review by rma-psmorris — APPROVED** [review]
approve

**Pull request merged**